appgate/syncer/operator.py

```diff
@@ -0,0 +1,204 @@
+    entity_type = entity.__class__.__qualname__
+
+    entity_dir = GIT_DUMP_DIR / f"{entity_type.lower()}-v{ctx.api_spec.api_version}"
+    entity_dir.mkdir(exist_ok=True)
+    entity_file = entity_dir / f"{entity.name.lower().replace(' ', '-')}.yaml"
+    dumped_entity = dump_entity(
+        EntityWrapper(entity), entity_type, f"v{ctx.api_spec.api_version}"
+    )
+
+    appgate_metadata = dumped_entity["spec"].get(APPGATE_METADATA_ATTRIB_NAME)
+    if appgate_metadata:
+        entity_passwords = appgate_metadata.get(APPGATE_METADATA_PASSWORD_FIELDS_FIELD)
+    dumped_entities.append(
+        yaml.safe_dump(dumped_entity, default_flow_style=False, sort_keys=True)
+    )
+    with entity_file.open("w") as f:
+        for i, de in enumerate(dumped_entities):
+            if i > 0:
+                f.write("---\n")
+            f.write(de)
+
+
+def print_configuration(ctx: GitOperatorContext):
+    log.info(
+        "[git-operator] Starting Git Syncer loop with the following configuration: "
+    )
+    log.info(
+        "[git-operator]     Target tags: %s",
+        ",".join(ctx.target_tags) if ctx.target_tags else "None",
+    )
+    log.info("[git-operator]     Log level: %s", ctx.log_level)
+    log.info("[git-operator]     Timeout: %s", ctx.timeout)
+    log.info(
+        "[git-operator]     Git repository: %s",
+        ctx.git_repository,
+    )
+    log.info("[git-operator]     Git vendor: %s", ctx.git_vendor)
+    log.info("[git-operator]     Git username: %s", ctx.git_username)
+    log.info("[git-operator]     Git base branch: %s", ctx.git_base_branch)
+    log.info("[git-operator]     Dry-run mode: %s", ctx.dry_run)
+
+
+async def git_operator(queue: Queue, ctx: GitOperatorContext) -> None:
+    entities: List[Entity_T] = []
+    error_events: List[AppgateEventError] = []
+
+    git: GitRepo = get_git_repository(ctx)
+    print_configuration(ctx)
+
+    while True:
+        try:
+            event: AppgateEvent = await asyncio.wait_for(
+                queue.get(), timeout=ctx.timeout
+            )
+
+            if isinstance(event, AppgateEventError):
+                error_events.append(event)
+                continue
+
+            log.info(
+                '[git-operator] Event: %s entity of type %s "%s"',
+                event.op,
+                event.entity.__class__.__qualname__,
+                event.entity.name,
+            )
+            entities.append(event.entity)
+
+        except asyncio.exceptions.TimeoutError:
+            if error_events:
+                for event_error in error_events:
+                    log.error(
+                        "[git-operator] Entity of type %s %s: %s",
+                        event_error.name,
+                        event_error.kind,
+                        event_error.error,
+                    )
+                sys.exit(1)
+
+            branch = f'{str(datetime.date.today())}.{time.strftime("%H-%M-%S")}'
+            git.checkout_branch(branch, ctx.dry_run)
```

Comment:
Since you already pass `ctx` to the Git constructor in line 150, you could probably have a `dry_run` field in the Git class instead of passing the boolean as a parameter.

Same for `git.commit_change()` `git.push_change()`, and `git.create_pull_requests`